pkg/kubelet/network/dns/dns.go

```diff
@@ -273,13 +273,40 @@ func parseResolvConf(reader io.Reader) (nameservers []string, searches []string,
 			}
 		}
 		if fields[0] == "options" {
-			options = fields[1:]
+			options = appendOptions(options, fields[1:]...)
 		}
 	}
 
 	return nameservers, searches, options, utilerrors.NewAggregate(allErrors)
 }
 
+// appendOptions appends options to the given list, but does not add duplicates.
+// append option will overwrite the previous one either in new line or in the same line.
+func appendOptions(options []string, newOption ...string) []string {
+	var optionMap = make(map[string]int)
```

Comment:
I think that this will be simpler if you do a map[string]string , key = optName and value = option, and just iterate  updating the map in order (new options last, so override the previous one), then you can just dump the values of the map

```go
	var optionMap = make(map[string]int)
	for _, option := range options {
		optName := strings.Split(option, ":")[0]
		optionMap[optName] = option
        }
	for _, option := range newOption {
		optName := strings.Split(option, ":")[0]
		optionMap[optName] = option
        }
        options := []string
        for _, v:= range optionMap {
         options = append(options,v)
          }

```

Reply:
It looks better. Updated.